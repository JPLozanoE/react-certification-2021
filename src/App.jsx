import React, { useEffect, useReducer } from 'react';
import { appReducer } from './state/appReducer';
import { AppContext } from './state/AppContext';
import Theme from './components/Layout/Theme';
import MainRouter from './routers/MainRouter';

const init = () => {
  return {
    isDarkTheme: true,
    search: '',
    selectedVideo: null,
    videos: [],
    auth: JSON.parse(localStorage.getItem('auth')) || {
      isLogged: false,
      displayName: '',
      uid: null,
    },
    favoriteVideos: JSON.parse(localStorage.getItem('favoriteVideos')) || [],
  };
};
export const App = () => {
  const [state, dispatch] = useReducer(appReducer, {}, init);

  useEffect(() => {
    localStorage.setItem('auth', JSON.stringify(state.auth));
  }, [state.auth]);

  return (
    <div>
      <AppContext.Provider value={{ state, dispatch }}>
        <Theme>
          <MainRouter />
        </Theme>
      </AppContext.Provider>
    </div>
  );
};
