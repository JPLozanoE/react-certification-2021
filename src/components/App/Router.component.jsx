import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import HomePage from '../../pages/Home';

import Layout from '../Layout';
import GlobalStyles from '../../__globalStyles';

function Router() {
  return (
    <BrowserRouter>
      <GlobalStyles />
      <Layout>
        <Switch>
          <Route exact path="/">
            <HomePage />
          </Route>
        </Switch>
      </Layout>
    </BrowserRouter>
  );
}

export default Router;
