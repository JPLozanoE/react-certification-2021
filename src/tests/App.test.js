/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render } from '@testing-library/react';
import { App } from '../App';

describe('<App/> Testing', () => {
  test('should render properly', () => {
    const wrapper = render(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
