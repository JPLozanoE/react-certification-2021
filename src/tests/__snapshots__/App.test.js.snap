// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> Testing should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <main
          class="sc-hKFxyN fOcJjF"
        >
          <div
            class="sc-kEqXSa fRhAbd topnav"
            id="myTopnav"
          >
            <a
              class="sc-iqAclL jRYrJJ"
              href="/"
            >
              Home
            </a>
            <div
              class="sc-fujyAs iyNEPm topnav"
              id="myTopnav"
            >
              <input
                class="sc-pNWdM cLsCj"
                placeholder="Enter your search..."
                value=""
              />
              <button
                class="sc-jrsJWt foBoLi"
                name="search"
              >
                Search
              </button>
            </div>
            <div
              class="sc-hBMUJo ldJdCL"
            >
              <p
                class="sc-crzoAE kbehzr"
                href="#about"
              >
                Dark mode
              </p>
              <div
                class="sc-fnVZcZ cvvNjz"
              >
                <div
                  class="sc-dIsUp bctaC"
                >
                  <input
                    checked=""
                    class="sc-ksluID jYamWL"
                    id="checkbox"
                    type="checkbox"
                  />
                  <label
                    class="sc-bqGGPW jLTksx"
                    for="checkbox"
                  />
                </div>
                <a
                  class="sc-iqAclL jRYrJJ"
                  href="/auth"
                >
                  Login
                </a>
              </div>
            </div>
          </div>
          <div
            class="sc-dlnjwi eueSIH"
          >
            <div
              class="sc-eCApnc coObtW"
            />
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <main
        class="sc-hKFxyN fOcJjF"
      >
        <div
          class="sc-kEqXSa fRhAbd topnav"
          id="myTopnav"
        >
          <a
            class="sc-iqAclL jRYrJJ"
            href="/"
          >
            Home
          </a>
          <div
            class="sc-fujyAs iyNEPm topnav"
            id="myTopnav"
          >
            <input
              class="sc-pNWdM cLsCj"
              placeholder="Enter your search..."
              value=""
            />
            <button
              class="sc-jrsJWt foBoLi"
              name="search"
            >
              Search
            </button>
          </div>
          <div
            class="sc-hBMUJo ldJdCL"
          >
            <p
              class="sc-crzoAE kbehzr"
              href="#about"
            >
              Dark mode
            </p>
            <div
              class="sc-fnVZcZ cvvNjz"
            >
              <div
                class="sc-dIsUp bctaC"
              >
                <input
                  checked=""
                  class="sc-ksluID jYamWL"
                  id="checkbox"
                  type="checkbox"
                />
                <label
                  class="sc-bqGGPW jLTksx"
                  for="checkbox"
                />
              </div>
              <a
                class="sc-iqAclL jRYrJJ"
                href="/auth"
              >
                Login
              </a>
            </div>
          </div>
        </div>
        <div
          class="sc-dlnjwi eueSIH"
        >
          <div
            class="sc-eCApnc coObtW"
          />
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
